### PR TITLE
feat(swizdb): add option to list keys, filterable by directory

### PR DESCRIPTION
**Pull request by flying-sausages**
## Description
Adds a listing feature to the swizdb

## Fixes issues: 
N/A

## Proposed Changes:
- Add `swizdb list [ key ]`

## Change Categories

- New feature 
- Change in `functions` 

## Checklist

- [ ] Docs have been made OR are not necessary
    - PR link: 
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas

### Architectures

|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|			|				|
| Bionic		|			|			|			|				|
| Buster		|			|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|	⚫️		|			|	⚫️		|	⚫️			|

### ✅❎ Passed
image

### CODE REVIEW

File: sources/functions/swizdb [hunk 1]
```diff
@@ -49,6 +42,19 @@ _getswizdb() {
...
+#Recursively lists keys
+# \
+# $1 = optional directory to refine
+# \
+# **Returns error code of the command that fails**
+_listswizdb() {
+    key="$1"
+    echo_log_only "List of keys stored under directory \"$key\""
+    pushd "$dbdir/$key" > /dev/null || return $? # Return error code of pushd, but stderr is not redirected on purpose.
+    find * -type f || return $?
```
Comment (flying-sausages) [inline sources/functions/swizdb:54]: I'm wondering if there's a better way to do this, but this is weirdly the simplest way to get what I want: just a line-separated list of files only without any prepending `./` or full path or anything like that

File: sources/functions/swizdb [hunk 2]
```diff
@@ -49,6 +42,19 @@ _getswizdb() {
...
+    popd > /dev/null || return
+}
+
```
Comment (userdocs) [inline sources/functions/swizdb:57]:
```suggestion
export -f _listswizdb
```

File: sources/functions/swizdb (hunk 1 above)
Comment (userdocs) [reply sources/functions/swizdb:54]:
```suggestion
    find "$dbdir/$key" -type f -exec basename {} \; | sort || return $?
```

Comment (userdocs) [reply sources/functions/swizdb:54]: don't think we need to pushd in to the dir

**Review by userdocs — APPROVED** [review]
it works. we tested it!

**Pull request merged**